pkg/sql/catalog/table_elements.go

```diff
@@ -154,12 +154,15 @@ type Index interface {
 	GetColumnName(columnOrdinal int) string
 	GetColumnDirection(columnOrdinal int) descpb.IndexDescriptor_Direction
 
-	ForEachColumnID(func(id descpb.ColumnID) error) error
-	ContainsColumnID(colID descpb.ColumnID) bool
+	CollectColumnIDs() TableColSet
+	CollectSecondaryStoredColumnIDs() TableColSet
+	CollectExtraColumnIDs() TableColSet
+	CollectCompositeColumnIDs() TableColSet
+
 	InvertedColumnID() descpb.ColumnID
 	InvertedColumnName() string
 
-	NumStoredColumns() int
+	NumSecondaryStoredColumns() int
```

Comment:
The idea behind this rename is that there soon might be a `NumPrimaryStoredColumns` as well as a `CollectPrimaryStoredColumnIDs`.